jerry-core/jerry.cpp

```diff
@@ -899,6 +900,66 @@ bool jerry_api_get_object_field_value (jerry_api_object_t *object_p,
 }
 
 /**
+ * Applies the given function to the every fields in the objects
+ *
+ * @return true, if object fields traversal was performed successfully, i.e.:
+ *                - no unhandled exceptions were thrown in object fields traversal;
+ *                - object fields traversal was stopped on callback that returned false;
+ *         false - otherwise,
+ *                 if getter of field threw a exception or unhandled exceptions were thrown during traversal;
+ */
+bool
+jerry_api_foreach_object_field (jerry_api_object_t *object_p, /**< object */
+                               jerry_object_field_foreach_t foreach_p, /**< foreach function */
+                               void *user_data_p) /**< user data for foreach function */
+{
+  jerry_assert_api_available ();
+
+  bool continuous = true;
+  for (ecma_property_t *property_p = ecma_get_property_list (object_p);
+       property_p != NULL && continuous;
+       property_p = ECMA_GET_POINTER (ecma_property_t, property_p->next_property_p))
+  {
```

Comment:
@idkiller, could you, please, update the property enumeration code like in commit 09b4d2071812b3212be80cadd016e407c1fc699b, as the #694 patch was landed?